Dynatrace/dynatrace-operator
# Improve metadata-enrichment test to use the different namespace-selectors

## Description



Adds more types of sample apps to the metadata-enrichment `Feature`.
So now we check if the 2 different namespace-selectors are respected AND if the annotations on the Pods are also respected.

## How can this be tested?


⚠️ Its a DRAFT because https://github.com/Dynatrace/dynatrace-operator/pull/3254 is needed it for it to work, otherwise it make the conversion webhook panic 😬 
⚠️ Due to a bug in the framework `make test/e2e/applicationmonitoring/metadataenrichment` WON'T work. [See issue](https://github.com/kubernetes-sigs/e2e-framework/issues/426)

I just commented out every other `Feature` in the `standard` test scenario, then ran it 🙃
